wsclient/src/main/java/org/hpccsystems/ws/client/antlr/ErrorListener.java

```diff
@@ -90,7 +91,18 @@ else if (e instanceof FailedPredicateException)
         {
             msg = "GENERAL_PARSE_ERROR," + e.getMessage();
         }
-        return msg + loc;
+        String loc = "";
+        if (e.getOffendingToken() != null)
+        {
+            loc = " at line " + e.getOffendingToken().getLine() + ", char " + e.getOffendingToken().getCharPositionInLine() + "-"
+                    + e.getOffendingToken().getCharPositionInLine()
+                    + (e.getOffendingToken().getStopIndex() - e.getOffendingToken().getStartIndex()) ;
+            
+        } else {
+            loc=" at line " + line + ", char " + charPositionInLine;
+        }
+        msg=msg + loc;
```

Comment:
I'm not very well versed on the java String operators, but on the surface it seems this operation would be more efficient as msg.append(loc), also, might be worthwhile to avoid this operation all together if loc.len == 0